clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4523,6 +4494,19 @@ void SYCLIntegrationFooter::addVarDecl(const VarDecl *VD) {
   SpecConstants.push_back(VD);
 }
 
+void SYCLIntegrationFooter::addTypeToCheckForDeviceCopyability(QualType QT) {
+  // FIXME: should this be an assertion instead?
+  if (!QT->isRecordType())
```

Comment:
Since you're testing for the record, when we hit the 'handleArrayType' this will end up getting an array type here.  The previous implementation in 'array' grabbed its element type and checked THAT.  

I don't think we need it to ensure 'record-type' as an assert unless you want to check if it is a record in your array handling code.